lib/service/service.go

```diff
@@ -2568,11 +2568,14 @@ func (process *TeleportProcess) initUploaderService() error {
 		return trace.Wrap(err)
 	}
 
-	log.Infof("starting upload completer service")
-
 	connectors := process.getConnectors()
 	var conn *Connector
 	for _, c := range connectors {
```

Comment:
@zmb3 this is what we talked about - in practice it seems to pick RoleInstance and go with it. I suppose that, depending on the order of the connectors, this could happen to most Teleport services.

Reply:
Ok, so I tested by forcing an agent to use `RoleInstance`, and the uploader does have the permissions it needs:

```
2023-05-18T10:03:03-06:00 DEBU [UPLOAD]    Scanned 1 uploads, started 1, found 0 corrupted in /Users/zmb/t/standlonenodedatadir/log/upload/streaming/default. filesessions/fileasync.go:270
2023-05-18T10:03:03-06:00 DEBU [UPLOAD]    Session upload completed. duration:26.65175ms session-id:ece63085-f291-47a4-9dfb-8890e598aa59 filesessions/fileasync.go:436
```

We should augment `TestLocalServiceRolesHavePermissionsForUploaderService` to ensure that it also verifies that `RoleInstance` can make these API calls. It's not quite as easy as adding this role to the test case since you have to construct the instance role separately. Let me know if you want to tackle that here or if we should do it separately.

Reply:
I've got the test running, wasn't that bad, but for the sake of backporting further than RoleMDM we can make it a separate PR.

Given your testing, should we instead make the uploader always use RoleInstance and simplify all of this? WDYT?

Reply:
Discussed offline, no changes to be made right now.